package/MDAnalysis/core/topology.py

```diff
@@ -348,3 +350,64 @@ def add_TopologyAttr(self, topologyattr):
         self.attrs.append(topologyattr)
         topologyattr.top = self
         self.__setattr__(topologyattr.attrname, topologyattr)
+
+    def to_json(self, filename):
+        """Write out Topology to JSON form.
+
+        Parameters
+        ----------
+        filename : str
+            File to write.
+
+        """
+        jstruct = {'n_atoms': self.n_atoms,
+                   'n_residues': self.n_residues,
+                   'n_segments': self.n_segments,
+                   'atoms->residues': self.tt.AR.tolist(),
+                   'residues->segments': self.tt.RS.tolist(),
+                   'topattrs': dict()}
+
+        # TODO: add warning that could not serialize some topology attributes
+        for topattr in self.attrs:
+
+            # these don't hold any data of their own
+            if isinstance(topattr, (Atomindices, Resindices, Segindices)):
+                continue
+
+            try:
+                jstruct['topattrs'][topattr.attrname] = topattr._to_json()
+            except AttributeError:
+                pass
+
+        # write out
+        with open(filename, 'w') as f:
```

Comment:
Could it be `util.anyopen`? This would allow to compress the json topology, but also to do funny things with `StreamIO` to interface this with other tools (in the same way as the interface with nglview writes a PDB file in a stream).